lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py

```diff
@@ -0,0 +1,429 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_containerinstance
+version_added: "2.5"
+short_description: Manage an Azure Container Instance.
+description:
+    - Create, update and delete an Azure Container Instance.
+
+options:
+    resource_group:
+        description:
+            - Name of resource group.
+        required: true
+    name:
+        description:
+            - The name of the container group.
+        required: true
+    os_type:
+        description:
+            - The OS type of containers.
+        choices:
+            - linux
+            - windows
+        default: linux
+    state:
+        description:
+            - Assert the state of the container instance. Use 'present' to create or update an container instance and 'absent' to delete it.
+        default: present
+        choices:
+            - absent
+            - present
+    ip_address:
+        description:
+            - The IP address type of the container group (default is 'none')
+        choices:
+            - public
+            - none
+        default: None
+    ports:
+        description:
+            - List of ports exposed within the container group.
+    location:
+        description:
+            - Valid azure location. Defaults to location of the resource group.
+        default: resource_group location
+    registry_login_server:
+        description:
+            - The container image registry login server.
+    registry_username:
+        description:
+            - The username to log in container image registry server.
+    registry_password:
+        description:
+            - The password to log in container image registry server.
+    containers:
+        description:
+            - List of containers.
+        suboptions:
+            name:
+                description:
+                    - The name of the container instance.
+                required: true
+            image:
+                description:
+                    - The container image name.
+                required: true
+            memory:
+                description:
+                    - The required memory of the containers in GB.
+                default: 1.5
+            cpu:
+                description:
+                    - The required number of CPU cores of the containers.
+                default: 1
+            ports:
+                description:
+                    - List of ports exposed within the container group.
+    force_update:
+        description:
+            - Force update of existing container instance.
```

Comment:
Probably expand on this a bit, maybe say if false then an existing container will not be changed and so on.